pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArithmeticFunctions.java

```diff
@@ -201,6 +204,34 @@ public static double truncate(double a) {
     return Math.signum(a) * Math.floor(Math.abs(a));
   }
 
+  @ScalarFunction
+  public static double rand() {
+    return ThreadLocalRandom.current().nextDouble();
+  }
+
+  @ScalarFunction
+  public static double rand(long seed) {
+    return deterministicRand(seed);
+  }
+
+  private static double deterministicRand(long seed) {
+    // The XOR with 0x5DEECE66DL is inspired by java.util.Random's initial LCG multiplier scramble,
+    // but unlike Random, we use a custom non-linear mix64 routine for diffusion and reproducible output.
+    // This approach is chosen to provide deterministic, well-diffused pseudo-random values, not to match Random's
```

Comment:
Are we not using the actual `Random` class here for the seeded random to reduce overhead and contention?

Reply:
yes mainly the contention